test-framework/emulator_backend.go

```diff
@@ -0,0 +1,159 @@
+	// TODO: maybe re-use interpreter? Only needed for value conversion
+	// TODO: Deal with imported/composite types
+	inter, err := newInterpreter()
+	if err != nil {
+		return interpreter.ScriptResult{
+			Error: err,
+		}
+	}
+
+	arguments := make([][]byte, 0, len(args))
+	for _, arg := range args {
+		exportedValue, err := runtime.ExportValue(arg, inter, interpreter.ReturnEmptyLocationRange)
+		if err != nil {
+			return interpreter.ScriptResult{
+				Error: err,
+			}
+		}
+
+		encodedArg, err := json.Encode(exportedValue)
+		if err != nil {
+			return interpreter.ScriptResult{
+				Error: err,
+			}
+		}
+
+		arguments = append(arguments, encodedArg)
+	}
+
+	result, err := e.blockchain.ExecuteScript([]byte(code), arguments)
+	if err != nil {
+		return interpreter.ScriptResult{
+			Error: err,
+		}
+	}
+
+	if result.Error != nil {
+		return interpreter.ScriptResult{
+			Error: result.Error,
+		}
+	}
+
+	value, err := runtime.ImportValue(inter, interpreter.ReturnEmptyLocationRange, result.Value, nil)
+	if err != nil {
+		return interpreter.ScriptResult{
+			Error: err,
+		}
+	}
+
+	return interpreter.ScriptResult{
+		Value: value,
+	}
+}
+
+// newBlockchain returns an emulator blockchain for testing.
+func newBlockchain(opts ...emulator.Option) *emulator.Blockchain {
+	b, err := emulator.NewBlockchain(
+		append(
+			[]emulator.Option{
+				emulator.WithStorageLimitEnabled(false),
+			},
+			opts...,
+		)...,
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	return b
+}
+
+// newInterpreter creates an interpreter instance needed for the value conversion.
+//
+func newInterpreter() (*interpreter.Interpreter, error) {
+	predeclaredInterpreterValues := stdlib.BuiltinFunctions.ToInterpreterValueDeclarations()
+	predeclaredInterpreterValues = append(predeclaredInterpreterValues, stdlib.BuiltinValues.ToInterpreterValueDeclarations()...)
+
+	return interpreter.NewInterpreter(
+		nil,
+		utils.TestLocation,
+		interpreter.WithStorage(interpreter.NewInMemoryStorage(nil)),
+		interpreter.WithPredeclaredValues(predeclaredInterpreterValues),
+		interpreter.WithImportLocationHandler(func(inter *interpreter.Interpreter, location common.Location) interpreter.Import {
+			switch location {
+			case stdlib.CryptoChecker.Location:
+				program := interpreter.ProgramFromChecker(stdlib.CryptoChecker)
+				subInterpreter, err := inter.NewSubInterpreter(program, location)
+				if err != nil {
+					panic(err)
+				}
+				return interpreter.InterpreterImport{
+					Interpreter: subInterpreter,
+				}
+
+			case stdlib.TestContractLocation:
+				program := interpreter.ProgramFromChecker(stdlib.TestContractChecker)
+				subInterpreter, err := inter.NewSubInterpreter(program, location)
+				if err != nil {
+					panic(err)
+				}
+				return interpreter.InterpreterImport{
+					Interpreter: subInterpreter,
+				}
+
+			default:
+				panic(errors.NewUnexpectedError("importing programs not implemented"))
```

Comment:
This is a little less ambiguous:
```suggestion
				panic(errors.NewUnexpectedError("importing of programs not implemented"))
```